buildkite/data_source_pipeline.go

```diff
@@ -2,70 +2,118 @@ package buildkite
+}
+
+func (*pipelineDatasource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_pipeline"
+}
+
+func (*pipelineDatasource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
 				Computed: true,
-				Type:     schema.TypeString,
 			},
-			"description": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"name": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The name of the pipeline.",
 			},
-			"name": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"default_branch": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The default branch to prefill when new builds are created or triggered.",
 			},
-			"repository": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"description": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The description of the pipeline.",
 			},
-			"slug": {
-				Required: true,
-				Type:     schema.TypeString,
+			"repository": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The git URL of the repository.",
 			},
-			"webhook_url": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"slug": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The slug of the pipeline.",
+			},
+			"webhook_url": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The Buildkite webhook URL that triggers builds on this pipeline.",
 			},
 		},
 	}
 }
 
-// ReadPipeline retrieves a Buildkite pipeline
-func dataSourcePipelineRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	// Warning or errors can be collected in a slice type
-	var diags diag.Diagnostics
+func (c *pipelineDatasource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state pipelineDataSourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
 
-	client := m.(*Client)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
-	orgPipelineSlug := fmt.Sprintf("%s/%s", client.organization, d.Get("slug").(string))
-	pipeline, err := getPipeline(client.genqlient, orgPipelineSlug)
+	orgPipelineSlug := fmt.Sprintf("%s/%s", c.client.organization, state.Slug.ValueString())
+
+	log.Printf("Obtaining pipeline with slug %s ...", orgPipelineSlug)
+	pipeline, err := getPipeline(c.client.genqlient, orgPipelineSlug)
 
 	if err != nil {
-		return diag.FromErr(err)
+		resp.Diagnostics.AddError(
+			"Unable to read pipeline",
+			fmt.Sprintf("Unable to read pipeline: %s", err.Error()),
+		)
+		return
 	}
 
 	if pipeline.Pipeline.Id == "" {
-		return diag.FromErr(errors.New("Pipeline not found"))
+		if err != nil {
```

Comment:
This condition is covered above. I think it should be removed here

Reply:
Multi err - in latest 👍